sandbox/lua/encoders/schema_carbon_line.lua

```diff
@@ -0,0 +1,120 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[=[
+Converts full Heka message contents to line protocol for Carbon Plaintext API
+Iterates through all of the dynamic fields to add as points (series), skipping
+any fields explicitly omitted using the `skip_fields` config option.  All
+dynamic fields in the Heka message are converted to separate points separated
+by newlines that are submitted to Carbon.
+
+Config:
+
+- name_prefix (string, optional, default nil)
+    String to use as the `name` key's prefix value in the generated line.
+    Supports :ref:`message field interpolation<sandbox_msg_interpolate_module>`.
+    `%{fieldname}`. Any `fieldname` values of "Type", "Payload", "Hostname",
+    "Pid", "Logger", "Severity", or "EnvVersion" will be extracted from the
+    the base message schema, any other values will be assumed to refer to a
+    dynamic message field. Only the first value of the first instance of a
+    dynamic message field can be used for name name interpolation. If the
+    dynamic field doesn't exist, the uninterpolated value will be left in the
+    name. Note that it is not possible to interpolate either the
+    "Timestamp" or the "Uuid" message fields into the name, those
+    values will be interpreted as referring to dynamic message fields.
+
+- name_prefix_delimiter (string, optional, default ".")
+    String to use as the delimiter between the name_prefix and the field
+    name.  This defaults to a "." to use Graphite naming convention.
+
+- skip_fields (string, optional, default nil)
+    Space delimited set of fields that should *not* be included in the
+    Carbon records being generated. Any fieldname values of "Type",
+    "Payload", "Hostname", "Pid", "Logger", "Severity", or "EnvVersion" will
+    be assumed to refer to the corresponding field from the base message
+    schema. Any other values will be assumed to refer to a dynamic message
+    field. The magic value "**all_base**" can be used to exclude base fields
+    from being mapped to the event altogether.
+
+- source_value_field (string, optional, default nil)
```

Comment:
I added this config option and associated functionality since I'm primarily using this encoder to take the output of data collection scripts, splitting messages from it by newline, parsing them with the Payload Regex Decoder and then parsing them to Carbon lines with this encoder.  This allows the encoder to work the original way (by sending multiple fields as separate Carbon lines) or the way I just described, which identifies a single field in the message that contains the value (e.g. "5.0") and adds it to the Carbon line.